### ISSUE TITLE
Add default headers or make them configurable

### ISSUE THREAD

**Issue by sbp-bvanb** [vouch/vouch-proxy#583]
Something like custom headers: https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#custom-headers

By default, the voucher should set some default headers in the response like Referrer-Policy

**Comment by bnfinet** [comment]
@sbp-bvanb thanks for the suggestion

Could you expand on this a bit?  What's your motivation here?  What problem are you trying to solve?

**Comment by sbp-bvanb** [comment]
A number of default security headers should be present in multiple layers, also on application level like:

			c.Response().Header().Set("Referrer-Policy", "no-referrer")
			c.Response().Header().Set("Strict-Transport-Security", "max-age=31536000; includeSubDomains; preload")
			c.Response().Header().Set("X-Content-Type-Options", "nosniff")
			c.Response().Header().Set("X-Frame-Options", "DENY")

Other application like Grafana and ArgoCD do this as well if a request is issues to their services.

**Comment by bnfinet** [comment]
@sbp-bvanb I'm of the opinion that such headers should be added via NGINX

https://nginx.org/en/docs/http/ngx_http_headers_module.html

There are many security related headers and each have their subtle nuances. 
https://cheatsheetseries.owasp.org/cheatsheets/HTTP_Headers_Cheat_Sheet.html
https://owasp.org/www-project-secure-headers/

I could see [improving the documentation](https://github.com/vouch/vouch-proxy/blob/master/README.md?plain=1#L167-L180) to highlight best practices.  Would you be interested in developing a PR to make that addition?

**Comment by sbp-bvanb** [comment]
In my opinion it should be resolved in the code as well. In our case we have an AWS ALB and the Nginx-ingress controller cannot configure the headers there. Instead of putting an AWS CloudFront everywhere, it should be resolved in all layers. I am willing to create a code and documentation change the upcoming weeks.

**Comment by bnfinet** [comment]
This shows a method of adding headers to the ingress controller.  Is that a workable solution for you?

https://stackoverflow.com/questions/54083179/how-can-i-correctly-setup-custom-headers-with-nginx-ingress#55574720

**Comment by sbp-bvanb** [comment]
Yes tried that today, but is not working in conjunction with an AWS ALB.
